### PR TITLE
INSTALL-944: Improve CI pipeline to make it scalable

### PR DESCRIPTION
**Pull request by lulivi**
### Summary

The current CI state of the examples repository consists in building all the examples in 4 build modes:

- Release build with dynamic library linking
- Release build with static library linking
- Debug build with dynamic library linking
- Debug build with static library linking

This PR make the Jenkinsfile simpler and adds some enhancements to allow building the examples in other platforms rather than Linux docker nodes.

### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -47,30 +67,64 @@ pipeline {
...
 
             environment {
                 RTI_INSTALLATION_PATH = "${env.WORKSPACE}"
+                VIRTUAL_ENV = "${env.WORKSPACE}/.venv"
             }
 
             stages {
+                stage('Select CMake utils version') {
+                    steps {
+                        script {
```
Comment (ManuelJNunez) [inline Jenkinsfile:96]: If we could get rid of the script blocks that would be great :)

File: Jenkinsfile [hunk 2]
```diff
@@ -47,30 +67,64 @@ pipeline {
...
+                                    error('An error ocurred installing Python dependencies')
+                                }
+
+                                withAWSCredentials {
+                                    withCredentials([
+                                        string(credentialsId: 's3-bucket', variable: 'RTI_AWS_BUCKET'),
+                                        string(credentialsId: 's3-path', variable: 'RTI_AWS_PATH'),
+                                    ]) {
+
+                                        if (runCommand(
```
[the remark this answers is not in the document]
Comment (ManuelJNunez) [reply Jenkinsfile:121]: Also if we do this, the error log won't be printed in the GitHub Check, only the error message which is not helpful.

File: Jenkinsfile (hunk 1 above)
Comment (lulivi) [reply Jenkinsfile]: It does not like that:
```
[GitHub Checks] GitHub check (name: Validation (Linux), status: in_progress) has been published.
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
WorkflowScript: 75: Method calls on objects not allowed outside "script" blocks. @ line 75, column 17.
                   nodeManager.runInsideExecutor() {
                   ^

WorkflowScript: 107: Method calls on objects not allowed outside "script" blocks. @ line 107, column 29.
                               nodeManager.runInsideExecutor() {
                               ^

WorkflowScript: 118: Method calls on objects not allowed outside "script" blocks. @ line 118, column 17.
                   nodeManager.runInsideExecutor() {
                   ^

3 errors
```

**Review by ManuelJNunez — APPROVED** [review]

**Review by ManuelJNunez — APPROVED** [review]

**Comment by ManuelJNunez** [comment]
- [ ] Should we merge to develop instead?

**Review by ManuelJNunez — CHANGES_REQUESTED** [review]
Answer my question before merging

**Review by ManuelJNunez — APPROVED** [review]

**Pull request merged**